lib/SIL/IR/SILPrinter.cpp

```diff
@@ -937,6 +938,21 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
         [&] { *this << ", "; });
   }
 
+  void printBranchTargets(const SILInstruction *inst) {
+    if (auto condBr = dyn_cast<CondBranchInst>(inst)) {
+      if (condBr->getTrueBB()->getDebugName().hasValue()) {
+        *this << ", true->" << condBr->getTrueBB()->getDebugName().getValue();
+      }
+      if (condBr->getFalseBB()->getDebugName().hasValue()) {
+        *this << ", false->" << condBr->getFalseBB()->getDebugName().getValue();
+      }
+    } else if (auto br = dyn_cast<BranchInst>(inst)) {
+      if (br->getDestBB()->getDebugName().hasValue()) {
+        *this << ", dest->" << br->getDestBB()->getDebugName().getValue();
+      }
+    }
```

Comment:
Could we generalize this for all terminators by casting the `inst` to a `TermInst` and iterating over the successors? We could then just print a generic name mapping like `bb1=<debug name>, bb2=<debug name>` for all of them. 

This would require readers to remember what each argument of the terminator corresponds to, but I believe it's just `cond_br` that requires remembering `true, false`, since the other terminators like `try_apply` get labels on their successors. Then again, we could also add `true` and `false` in the print-out of `cond_br`'s two operands.

Reply:
Cool sounds good I'll give it a shot

Reply:
Good idea. I admit to occasionally second guessing which cond_br successor is true, so those labels would be somewhat useful

Reply:
Done:

```
switch_enum %14 : $Optional<MyDistActor>, case #Optional.some!enumelt: bb4, case #Optional.none!enumelt: bb3 // id: %19, bb0->resolvedBB, bb1->makeProxyBB
```

So for branches we name them, and there's a fallback to just print bb### for others